bulk.go

```diff
@@ -0,0 +1,111 @@
+package opensearchtools
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+)
+
+// BulkRequest wraps the functionality of [opensearchapi.Bulk] by supporting request body creation.
+// An empty BulkRequest will fail to execute. At least one Action is required to be added.
+//
+// For more details see https://opensearch.org/docs/latest/api-reference/document-apis/bulk/
+type BulkRequest struct {
+	Actions []Action
+	Refresh Refresh
+}
+
+// NewBulkRequest instantiates an empty BulkRequest
+func NewBulkRequest() *BulkRequest {
```

Comment:
Since at least 1 action is required for a request to be valid this constructor should accept one as a parameter
```
func NewBulkRequest(action Action, addtl ...Action) *BulkRequest {
   ...
}
```

Reply:
So then if people have a slice of actions it'd be `NewBulkRequest(actions[0], actions[1:]...)`, feels a little funky.

Reply:
That's fair.  I was really thinking out loud for how to accomplish the _pit of success_.  Looks like your latest refactor made this moot anyway.